pkg/service/pipeline.go

```diff
@@ -962,6 +963,128 @@ func (s *service) triggerPipeline(
 	return s.getOutputsAndMetadata(ctx, pipelineTriggerID, r, returnTraces)
 }
 
+func (s *service) triggerPipelineWithStream(
+	ctx context.Context,
+	ns resource.Namespace,
+	r *datamodel.Recipe,
+	isAdmin bool,
+	pipelineID string,
+	pipelineUID uuid.UUID,
+	pipelineReleaseID string,
+	pipelineReleaseUID uuid.UUID,
+	pipelineData []*pipelinepb.TriggerData,
+	pipelineTriggerID string,
+	returnTraces bool,
+	stream chan<- TriggerResult) error {
+
+	logger, _ := logger.GetZapLogger(ctx)
+
+	memoryKey, err := s.preTriggerPipeline(ctx, isAdmin, ns, r, pipelineTriggerID, pipelineData)
+	if err != nil {
+		return err
+	}
+	defer recipe.Purge(ctx, s.redisClient, pipelineTriggerID)
+
+	workflowOptions := client.StartWorkflowOptions{
+		ID:                       pipelineTriggerID,
+		TaskQueue:                worker.TaskQueue,
+		WorkflowExecutionTimeout: time.Duration(config.Config.Server.Workflow.MaxWorkflowTimeout) * time.Second,
+		RetryPolicy: &temporal.RetryPolicy{
+			MaximumAttempts: config.Config.Server.Workflow.MaxWorkflowRetry,
+		},
+	}
+
+	userUID := uuid.FromStringOrNil(resource.GetRequestSingleHeader(ctx, constant.HeaderUserUIDKey))
+
+	if userUID == uuid.Nil {
+		logger.Debug("triggerPipelineWithStream: failed to get userUID from request header")
+	}
+
+	we, err := s.temporalClient.ExecuteWorkflow(
+		ctx,
+		workflowOptions,
+		"TriggerPipelineWorkflow",
+		&worker.TriggerPipelineWorkflowParam{
+			BatchSize:        len(pipelineData),
+			MemoryStorageKey: memoryKey,
+			SystemVariables: recipe.SystemVariables{
+				PipelineID:          pipelineID,
+				PipelineUID:         pipelineUID,
+				PipelineReleaseID:   pipelineReleaseID,
```

Comment:
This was added recently, can you add the requester UID as well as in [here](https://github.com/instill-ai/pipeline-backend/blob/main/pkg/service/pipeline.go#L1021)? It will determine the credit owner on Cloud. Thanks! 🙇